sample_app/main_rtk.c

```diff
@@ -264,15 +193,57 @@ int main(void)
    appdata.main_events = os_event_create();
    appdata.main_timer = os_timer_create(TICK_INTERVAL_US, main_timer_tick, NULL, false);
 
-   printf("\n** Profinet demo application ready to start **\n");
+   printf("\n** Profinet sample application **\n");
    if (appdata.arguments.verbosity > 0)
    {
       printf("Number of slots:     %u (incl slot for DAP module)\n", PNET_MAX_MODULES);
       printf("P-net log level:     %u (DEBUG=0, ERROR=3)\n", LOG_LEVEL);
       printf("App verbosity level: %u\n", appdata.arguments.verbosity);
       printf("Ethernet interface:  %s\n", appdata.arguments.eth_interface);
+      printf("Station name:        %s\n\n", appdata.arguments.station_name);
    }
 
+   app_adjust_stack_configuration(&pnet_default_cfg);
+   strcpy(pnet_default_cfg.im_0_data.order_id, "12345");
+   strcpy(pnet_default_cfg.im_0_data.im_serial_number, "00001");
+   pnet_default_cfg.ip_addr.a = ip4_addr1 (&netif_default->ip_addr);
+   pnet_default_cfg.ip_addr.b = ip4_addr2 (&netif_default->ip_addr);
+   pnet_default_cfg.ip_addr.c = ip4_addr3 (&netif_default->ip_addr);
+   pnet_default_cfg.ip_addr.d = ip4_addr4 (&netif_default->ip_addr);
+   pnet_default_cfg.ip_mask.a = ip4_addr1 (&netif_default->netmask);
+   pnet_default_cfg.ip_mask.b = ip4_addr2 (&netif_default->netmask);
+   pnet_default_cfg.ip_mask.c = ip4_addr3 (&netif_default->netmask);
+   pnet_default_cfg.ip_mask.d = ip4_addr4 (&netif_default->netmask);
+   pnet_default_cfg.ip_gateway.a = pnet_default_cfg.ip_addr.a;
+   pnet_default_cfg.ip_gateway.b = pnet_default_cfg.ip_addr.b;
+   pnet_default_cfg.ip_gateway.c = pnet_default_cfg.ip_addr.c;
+   pnet_default_cfg.ip_gateway.d = 1;
+   memcpy (pnet_default_cfg.eth_addr.addr, netif_default->hwaddr, sizeof(pnet_ethaddr_t));
+   strcpy(pnet_default_cfg.station_name, gp_appdata->arguments.station_name);
+   pnet_default_cfg.cb_arg = (void*)gp_appdata;
+
+   g_net = pnet_init(gp_appdata->arguments.eth_interface, TICK_INTERVAL_US, &pnet_default_cfg);
+   if (g_net == NULL)
+   {
+      printf("Failed to initialize p-net application.\n");
+      return -1;
+   }
+
+   app_plug_dap(g_net, gp_appdata);
+
+   if (gp_appdata->arguments.verbosity > 0)
+   {
+      printf("Initialized p-net samople application.\n");
+      printf("Waiting for connect request from IO-controller\n\n");
+   }
+
+   if (gp_appdata->init_done == true)
```

Comment:
This can maybe be removed? Probably there to stop the timer if it was already running due to a restart, but that's no longer possible?

Reply:
Done